src/univariate/discrete/poissonbinomial.jl

```diff
@@ -141,8 +141,22 @@ function poissonbinomial_pdf_fft(p::AbstractArray)
             x[n + 1 - l + 1] = conj(x[l + 1])
         end
     end
-    fft!(x)
-    [max(0, real(xi)) for xi in x]
+    [max(0, real(xi)) for xi in _dft(x)]
+end
+
+# A simple implementation of a DFT to avoid introducing a dependency
+# on an external FFT package just for this one distribution
+if VERSION >= v"0.7.0-DEV.602"
+    function _dft(x::Vector{T}) where T
+        n = length(x)
+        y = fill!(copy(x), zero(complex(float(T))))
```

Comment:
Please look at this one again. I'm pretty sure it is not doing what you intended it to do. Furthermore, there is no reason to initialize the memory since that is done in the loop below.